### DISCUSSION TITLE
on_group_join and on_group_remove?

### DISCUSSION THREAD

**Discussion by thepeoplescoder** [Rapptz/discord.py#9732] [Q&A] [answered]
I apologize if this is a stupid question, but I can't seem to find it anywhere on Google or in the Documentation (I thought this would have been annotated in the Removals section, but it wasn't):

I noticed that the **on_group_remove** and **on_group_join** events were removed in discord.py >= 2.x, but I couldn't seem to find the rationale for it.  Did **Discord**'s internal API change?

In all fairness, it has been a number of years since I've worked discord.py; the last version I worked with was 1.7.3, so I'm aware that the project has moved forward in the past couple of years.  I'm currently using discord.py 2.3.2.

**Comment by Rapptz** [comment, marked as answer]
Those were for userbots which are no longer supported.

In the future they might come back depending on how app command contexts/user apps turn out but I'm not sure you'll get those events in those cases.
